mlir/lib/Dialect/Rock/Transforms/GemmLinalgSplitkNormalizationPass.cpp

```diff
@@ -90,52 +91,49 @@ static LogicalResult
 rewriteLinalgForSplitK(func::FuncOp &func,
                        BufferDependencyAnalysis &bufferDeps) {
   IRRewriter rewriter(func->getContext());
-  SmallVector<linalg::GenericOp> genericOps;
-  func.walk([&genericOps](linalg::GenericOp genericOp) {
-    genericOps.push_back(genericOp);
-  });
-  const auto &writersTable = bufferDeps.getWritersTable();
-  LLVM_DEBUG(llvm::dbgs() << "Found " << genericOps.size()
-                          << " linalg::GenericOp\n");
-
-  for (linalg::GenericOp op : genericOps) {
-    SmallVector<Value> gemmOut;
-    SmallVector<int64_t> splitKFactors;
-    SmallVector<GemmFeatures> features;
-    for (auto operand : op->getOperands()) {
-      auto genericOpInputAlloc = findMemrefAlloc(operand);
-      if (succeeded(genericOpInputAlloc)) {
-        if (writersTable.contains(genericOpInputAlloc.value())) {
-          for (OpOperand *op : writersTable.at(genericOpInputAlloc.value())) {
-            if (auto gemm = dyn_cast<GemmOp>(op->getOwner())) {
-              const int64_t splitKFactor = gemm.getParams()->getSplitKFactor();
-              LLVM_DEBUG(llvm::dbgs()
-                         << "splitKFactor=" << splitKFactor << "\n");
-              if (splitKFactor > 1) {
-                gemmOut.push_back(genericOpInputAlloc.value());
-                splitKFactors.push_back(splitKFactor);
-                features.push_back(gemm.getFeatures());
-              }
-            }
-          }
-        }
-      }
+  SmallVector<GemmOp> gemmOps;
+
+  func.walk([&](GemmOp gemmOp) {
+    int64_t splitKFactor = gemmOp.getParams()->getSplitKFactor();
+    if (splitKFactor > 1) {
+      gemmOps.push_back(gemmOp);
     }
-    assert(gemmOut.empty() || gemmOut.size() == 1);
-    assert(gemmOut.size() == splitKFactors.size());
-    assert(gemmOut.size() == features.size());
-    if (gemmOut.size() == 1) {
+  });
+  if (gemmOps.size() > 1)
+    return failure();
```

Comment:
```suggestion
  if (gemmOps.size() != 1)
    return failure();
```

and then remove If condition later that checks `gemmOps.size() == 1`

Reply:
We don't want to fail if there are no gemms in the kernel.